### ISSUE TITLE
Regression: Add "else" branches support for TypeIs

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#7837]
`pyright` version 1.1.361 does not match the description of [PEP 742](https://peps.python.org/pep-0742/) for else branches.

Is it possible to add support for these in `isinstance`, so that the new version of `TypeIs` will also support them?

From PEP: "TypeIs can narrow the type in both the if and else branches of a conditional."

"When a TypeIs function returns False, type checkers can narrow the type of the variable to exclude the TypeIs type."

**Comment by christian-hnz** [comment]
IMO this was working prior to version 1.1.361. At least, the following example from PEP 742 was working (with pyright 1.1.360).
```
from typing import assert_type
from typing_extensions import TypeIs

def is_str(x: object) -> TypeIs[str]:
    return isinstance(x, str)

def f(x: str | int) -> None:
    if is_str(x):
        assert_type(x, str)
    else:
        assert_type(x, int)
```

**Comment by christian-hnz** [comment]
I suspect this item in _Behavioral changes_ caused the issue:
* Changed TypeIs to use the same logic as isinstance type narrowing logic for consistency.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.362, which I just published.

**Comment by christian-hnz** [comment]
Thanks for fixing! Just updated to 1.1.362 and now it works as expected.
